docs/source/considerations.rst

```diff
@@ -93,17 +93,38 @@ In fact, the only difference is the binding from Alice, in which we are using
 a lambda function for the handler.
 
 
+.. _shutting_down:
+
 Shutting down
 =============
 
-Although not covered in the examples until now (because many times you just
-want the multi-agent system to run forever until, perhaps, an event occurs),
-it is possible to actively kill the system using proxies:
+Sometimes, we will want to manually shut down a single agent or even a group
+of agents. In this section, we will see how to do it.
+
+Shutting down agents
+--------------------
+
+If we want to end the execution of a specific agent in our system, we can do
+it by calling the `agent.shutdown` method:
 
 .. literalinclude:: ../../examples/shutdown.py
 
-.. note:: Shutting down the nameserver will result in all agents registered in
-   the name server being shut down as well.
+Shutting down name servers
+--------------------------
+
+We learned earlier that name servers run just like any other agent in our
+system, even though with specific functionality.
+
+Shutting down a name server will result in all agents registered in that name
+server being shut down as well. This allows us to easily shutdown groups of
+agents at the same time, by registering them in different name servers when
+created.
+
+.. literalinclude:: ../../examples/various_nameservers.py
+
+.. note:: As seen in the example above, that two agents are registered on
+   different name servers does not mean that we cannot stablish connections
+   between them.
```

Comment:
Definitely, the examples directory can hold many examples, even complex cases. Only not all need to be shown (maybe just linked) from the documentation. Dispatcher and ventilator are not displayed in the documentation for example (I think). Anyway, if the purpose of that example is to show the shutdown of a group of agents, try to constrain the example (i.e.: avoid, if possible, timers, pub-sub, topics...). :wink: